internal/saucecloud/retry/saucereportretrier_test.go

```diff
@@ -0,0 +1,168 @@
+package retry
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/cypress"
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/saucereport"
+	"github.com/saucelabs/saucectl/internal/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+type StubProjectUploader struct {
+}
+
+func (f *StubProjectUploader) UploadStream(filename, description string, reader io.Reader) (storage.Item, error) {
+	return storage.Item{
+		ID:   "fakeid",
+		Name: "fake name",
+	}, nil
+}
+
+func (f *StubProjectUploader) Download(id string) (io.ReadCloser, int64, error) {
+	return nil, 0, nil
+}
+
+func (f *StubProjectUploader) DownloadURL(url string) (io.ReadCloser, int64, error) {
+	return nil, 0, nil
+}
+
+func (f *StubProjectUploader) List(opts storage.ListOptions) (storage.List, error) {
+	return storage.List{}, nil
+}
+
+type StubVDCJobReader struct {
+	SauceReport saucereport.SauceReport
+}
+
+func (f *StubVDCJobReader) ReadJob(ctx context.Context, id string, realDevice bool) (job.Job, error) {
+	return job.Job{}, nil
+}
+
+func (f *StubVDCJobReader) PollJob(ctx context.Context, id string, interval, timeout time.Duration, realDevice bool) (job.Job, error) {
+	return job.Job{}, nil
+}
+
+func (f *StubVDCJobReader) GetJobAssetFileNames(ctx context.Context, jobID string, realDevice bool) ([]string, error) {
+	return []string{}, nil
+}
+
+func (f *StubVDCJobReader) GetJobAssetFileContent(ctx context.Context, jobID, fileName string, realDevice bool) ([]byte, error) {
+	body, _ := json.Marshal(f.SauceReport)
+	return []byte(body), nil
+}
+
+type FakeProject struct {
```

Comment:
This project does nothing, so likely not a fake 💭 

```suggestion
type StubProject struct {
```